File: Easy/645. Set Mismatch/solution.cs
public class Solution {
    public int[] FindErrorNums(int[] nums) {
        var freq = new Dictionary<int, int>();
        
        foreach(int n in nums){
            if (!freq.ContainsKey(n))
                freq[n] = 1;
            else
                freq[n]++;
        }

        int missing = 0, duplicate = 0;
        for (int i = 1; i <= nums.Length; i++){
            if (freq.ContainsKey(i)){
                if (freq[i] == 2)
                    duplicate = i;
            }
            else{
                missing = i;
            }
        }

        int[] ans = {duplicate, missing};
        return ans;
    }
}

File: Easy/645. Set Mismatch/solution.cpp
class Solution {
public:
    vector<int> findErrorNums(vector<int>& nums) {
        int N = nums.size();
        vector<int> freq(N + 1, 0);

        for (int n : nums){
            freq[n]++;
        }

        vector<int> result(2);
        for (int i = 1; i <= N; i++){
            if (freq[i] == 2){
                result[0] = i;
            }
            else if (freq[i] == 0){
                result[1] = i;
            }
        }
        return result;
    }
};